src/linear_algebra/quadratic_form.lean

```diff
@@ -788,3 +789,229 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
+  refine nonempty.intro _,
+  convert Q.isometry_of_is_basis hv₂,
+  ext w,
+  rw [isometry_of_is_Ortho_apply Q hv₂ hv₁, weighted_sum_squares_apply],
+  refine finset.sum_congr rfl _,
+  intros, rw ← mul_assoc,
+end
+
+lemma equivalent_weighted_sum_squares_of_nondegenerate
+  (Q : quadratic_form K V) (hQ : (associated Q).nondegenerate) :
+  ∃ w : fin (finite_dimensional.finrank K V) → K,
+    equivalent Q (weighted_sum_squares w) :=
+let ⟨w, _, hw₂⟩ := Q.equivalent_weighted_sum_squares_of_nondegenerate' hQ in ⟨w, hw₂⟩
+
+section complex
+
+/-- The weighted sum of squares on the complex numbers as a quadratic form is equivalent
+to the sum of squares, i.e. `weighted_sum_squares` with weight `λ i : ι, 1`. -/
+noncomputable def isometry_sum_squares [decidable_eq ι] (w : ι → ℂ) (hw : ∀ i : ι, w i ≠ 0) :
+  isometry (weighted_sum_squares w : quadratic_form ℂ (ι → ℂ))
+           (weighted_sum_squares (λ _, 1 : ι → ℂ)) :=
+begin
+  have hw' : ∀ i : ι, (w i) ^ - (1 / 2 : ℂ) ≠ 0,
+  { intros i hi,
+    exact hw i ((complex.cpow_eq_zero_iff _ _).1 hi).1 },
+  convert (weighted_sum_squares w : quadratic_form ℂ (ι → ℂ)).isometry_of_is_basis
+    (is_basis.smul_of_invertible (pi.is_basis_fun ℂ ι) (λ i, field.invertible (hw' i))),
+  ext1 v,
+  rw [isometry_of_is_basis_apply, weighted_sum_squares_apply, weighted_sum_squares_apply],
+  refine sum_congr rfl (λ j hj, _),
+  have hsum : (∑ (i : ι), v i • w i ^ - (1 / 2 : ℂ) •
+    (linear_map.std_basis ℂ (λ (i : ι), ℂ) i) 1) j =
+    v j • w j ^ - (1 / 2 : ℂ),
+  { rw [sum_apply, sum_eq_single j, linear_map.std_basis_apply, pi.smul_apply, pi.smul_apply,
+        function.update_same, smul_eq_mul, smul_eq_mul, smul_eq_mul, mul_one],
+    intros i _ hij,
+    rw [linear_map.std_basis_apply, pi.smul_apply, pi.smul_apply, function.update_noteq hij.symm,
+        pi.zero_apply, smul_eq_mul, smul_eq_mul, mul_zero, mul_zero],
+    intro hj', exact false.elim (hj' hj) },
+  rw [hsum, smul_eq_mul],
+  suffices : 1 * v j * v j =  w j ^ - (1 / 2 : ℂ) * w j ^ - (1 / 2 : ℂ) * w j * v j * v j,
+  { rw [← mul_assoc, this], ring },
+  rw [← complex.cpow_add _ _ (hw j), show - (1 / 2 : ℂ) + - (1 / 2) = -1, by ring,
+      complex.cpow_neg_one, inv_mul_cancel (hw j)],
+end .
+
+/-- A nondegenerate quadratic form on the complex numbers is equivalent to
+the sum of squares, i.e. `weighted_sum_squares` with weight `λ i : ι, 1`. -/
+theorem equivalent_sum_squares {M : Type*} [add_comm_group M] [module ℂ M]
+  [finite_dimensional ℂ M] (Q : quadratic_form ℂ M) (hQ : (associated Q).nondegenerate) :
+  equivalent Q (weighted_sum_squares (λ _, 1 : fin (finite_dimensional.finrank ℂ M) → ℂ)) :=
+let ⟨w, hw₁, hw₂⟩ := Q.equivalent_weighted_sum_squares_of_nondegenerate' hQ in
+  nonempty.intro $ (classical.choice hw₂).trans (isometry_sum_squares w hw₁)
+
+end complex
+
+section real
+
+/-- The sign function that maps negative real numbers to -1 and nonnegative numbers to 1. -/
+noncomputable def sign (r : ℝ) : ℝ := if r < 0 then -1 else 1
```

Comment:
I'm not sure if this exists elsewhere (I've only found `int.sign`) nor where it should go. I feel that it shouldn't go in `real.basic` since it feels out of place there.